src/app/containers/SignupContainer/useSignup.js

```diff
@@ -27,7 +27,8 @@ import {
 import { getPlan, PLAN, VPN_PLANS, PLAN_BUNDLES } from './plans';
```

Comment:
Since you are touching `useSignup` here, can you add comments to explain a bit the logic used please and why you are doing some stuff?

Reply:
I am not changing logic for signup, just moved around some code, to reduce code repetition. Mattias asked to make the signup a little more robust with incorrect coupons, so the signup will always try to use first possible valid data for plan/cycle/currency.
1) From initial model passed to signup
2) From coupon (first from history, then params)
3) Default: PLUS plan + 12 months + USD
So if invalid coupon or plan was passed it will always have something to fallback to. Previously, it didn't check coupons (assumed they were always correct)

Reply:
If you meant to put comments in code about how signup works in general, I think it would be better to create a wiki page with that. There are insane amount of rules, and most of them came up when building the signup, so the code is not very clear.